WcaOnRails/app/webpacker/components/Panel/PanelTemplate.jsx

```diff
@@ -3,21 +3,39 @@ import {
   Dropdown,
   Grid, Header, Icon, Menu, Segment,
 } from 'semantic-ui-react';
+import useHash from '../../lib/hooks/useHash';
 
 export default function PanelTemplate({ heading, sections }) {
-  const [selectedMenu, setSelectedMenu] = React.useState(0);
-  const SelectedComponent = React.useMemo(() => sections.find(
-    (_, index) => index === selectedMenu,
-  ).component, [sections, selectedMenu]);
+  const [hash, setHash] = useHash();
 
-  function menuClickHandler(index) {
-    const section = sections[index];
+  const selectedMenu = React.useMemo(() => (hash ? sections.findIndex(
+    (section) => section.id === hash,
+  ) : 0), [hash, sections]);
+
+  if (selectedMenu === -1) {
+    setHash(sections[0].id);
+  }
```

Comment:
Not a big fan of introducing a forced re-render by manually setting the hash here.

Can you just assume that "no hash" refers to the first available section and call it a day? You can start setting the hash as soon as the user actually clicks something, and otherwise just let React happily render the panel with a default selection without enforcing any hash.

Reply:
I feel that won't look clean because none of the components in panel page is a 'home' page. Everything is independent. So, I felt the URL should have the ID as hash. Reason is, `/panel/wfc` will open `/panel/wfc/dues-export` but if the URL doesn't say what it is for dues-redirect, but it does say for other components I feel it won't look good. Can you please consider/accept this as an exception?